### PR TITLE
feat(yaml,pdb): Litmus experiment to verify PDB of cspc pool pods

### PR DESCRIPTION
**Pull request by gprasath**
Signed-off-by: gprasath <giridhara.prasad@mayadata.io>

**What this PR does / why we need it**:
- It verifies whether CSPC pool pods are honoring the pod disruption budget created while cstor volume provisioning.
- Prerequisites is CSPC pool and volume should be provisioned on the cluster.
- Drains couple of nodes and validates if the PDB is not violated.
Issue: #171 
**Checklist**

* [x] Does this PR have a corresponding GitHub issue?
* [x] Have you included relevant README for the chaoslib/experiment with details?
* [x] Have you added debug messages where necessary? 
* [x] Have you added task comments where necessary? 
* [x] Have you tested the changes for possible failure conditions?
* [x] Have you provided the positive & negative test logs for the litmusbook execution?
* [ ] Does the litmusbook ensure idempotency of cluster state?, i.e., is cluster restored to original state?
* [ ] Have you used non-shell/command modules for Kubernetes tasks?
* [ ] Have you (jinja) templatized custom scripts that is run by the litmusbook, if any? 
* [ ] Have you (jinja) templatized Kubernetes deployment manifests used by the litmusbook, if any?
* [ ] Have you reused/created util functions instead of repeating tasks in the litmusbook?
* [ ] Do the artifacts follow the appropriate directory structure? 
* [ ] Have you isolated storage (eg: OpenEBS) specific implementations, checks? 
* [ ] Have you isolated platform (eg: baremetal kubeadm/openshift/aws/gcloud) specific implementations, checks?  
* [x] Are the ansible facts well defined? Is the scope explicitly set for playbook & included utils?
* [x] Have you ensured minimum/careful usage of shell utilities (awk, grep, sed, cut, xargs etc.,)?
* [ ] Can the limtusbook be executed both from within & outside a container (configurable paths, no hardcode)?
* [ ] Can you suggest the minimal resource requirements for the litmusbook execution?
* [x] Does the litmusbook job artifact carry comments/default options/range for the ENV tunables?
* [x] Has the litmusbooks been linted? 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by gprasath** [comment]
Test status.
```
PLAYBOOK: test.yml *************************************************************
1 plays in ./experiments/chaos/cspc_pod_disruption_budget/test.yml

PLAY [localhost] ***************************************************************
2020-03-03T07:44:02.629591 (delta: 0.055589)         elapsed: 0.055589 ******** 
=============================================================================== 

TASK [Gathering Facts] *********************************************************
task path: /experiments/chaos/cspc_pod_disruption_budget/test.yml:2
2020-03-03T07:44:02.641893 (delta: 0.012267)         elapsed: 0.067891 ******** 
ok: [127.0.0.1]
META: ran handlers

TASK [include_tasks] ***********************************************************
task path: /experiments/chaos/cspc_pod_disruption_budget/test.yml:13
2020-03-03T07:44:10.437922 (delta: 7.796007)         elapsed: 7.86392 ********* 
included: /utils/fcm/create_testname.yml for 127.0.0.1

TASK [Record test instance/run ID] *********************************************
task path: /utils/fcm/create_testname.yml:3
2020-03-03T07:44:10.528346 (delta: 0.08998)         elapsed: 7.954344 ********* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Construct testname appended with runID] **********************************
task path: /utils/fcm/create_testname.yml:7
2020-03-03T07:44:10.592354 (delta: 0.063939)         elapsed: 8.018352 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
task path: /experiments/chaos/cspc_pod_disruption_budget/test.yml:15
2020-03-03T07:44:10.650128 (delta: 0.057701)         elapsed: 8.076126 ******** 
included: /utils/fcm/update_litmus_result_resource.yml for 127.0.0.1

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
task path: /utils/fcm/update_litmus_result_resource.yml:3
2020-03-03T07:44:10.748811 (delta: 0.098633)         elapsed: 8.174809 ******** 
changed: [127.0.0.1] => {"changed": true, "checksum": "1c91874bb92ef6c2fee119251c322fea5cdfe308", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "ff8b694dd0164f313dbe09d9f9aa78a6", "mode": "0644", "owner": "root", "size": 453, "src": "/root/.ansible/tmp/ansible-tmp-1583221450.8-262295180154453/source", "state": "file", "uid": 0}

TASK [Analyze the cr yaml] *****************************************************
task path: /utils/fcm/update_litmus_result_resource.yml:14
2020-03-03T07:44:11.439516 (delta: 0.690639)         elapsed: 8.865514 ******** 
changed: [127.0.0.1] => {"changed": true, "cmd": "cat litmus-result.yaml", "delta": "0:00:00.740263", "end": "2020-03-03 07:44:12.496426", "rc": 0, "start": "2020-03-03 07:44:11.756163", "stderr": "", "stderr_lines": [], "stdout": "---\napiVersion: litmus.io/v1alpha1\nkind: LitmusResult\nmetadata:\n\n  # name of the litmus testcase\n  name: cspc-pod-disruption-budget \nspec:\n\n  # holds information on the testcase\n  testMetadata:\n    app:  \n    chaostype: cspc-pod-disruption-budget \n\n  # holds the state of testcase,  manually updated by json merge patch\n  # result is the useful value today, but anticipate phase use in future \n  testStatus: \n    phase: in-progress  \n    result: none ", "stdout_lines": ["---", "apiVersion: litmus.io/v1alpha1", "kind: LitmusResult", "metadata:", "", "  # name of the litmus testcase", "  name: cspc-pod-disruption-budget ", "spec:", "", "  # holds information on the testcase", "  testMetadata:", "    app:  ", "    chaostype: cspc-pod-disruption-budget ", "", "  # holds the state of testcase,  manually updated by json merge patch", "  # result is the useful value today, but anticipate phase use in future ", "  testStatus: ", "    phase: in-progress  ", "    result: none "]}

TASK [Apply the litmus result CR] **********************************************
task path: /utils/fcm/update_litmus_result_resource.yml:17
2020-03-03T07:44:12.551822 (delta: 1.112251)         elapsed: 9.97782 ********* 
ok: [127.0.0.1] => {"changed": false, "failed_when_result": false, "method": "patch", "result": {"apiVersion": "litmus.io/v1alpha1", "kind": "LitmusResult", "metadata": {"creationTimestamp": "2020-03-03T01:35:23Z", "generation": 3, "name": "cspc-pod-disruption-budget", "resourceVersion": "254400", "selfLink": "/apis/litmus.io/v1alpha1/litmusresults/cspc-pod-disruption-budget", "uid": "33e75fd1-5c14-43cc-9486-2d74fbf02103"}, "spec": {"testMetadata": {"chaostype": "cspc-pod-disruption-budget"}, "testStatus": {"phase": "in-progress", "result": "none"}}}}

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
task path: /utils/fcm/update_litmus_result_resource.yml:27
2020-03-03T07:44:13.616569 (delta: 1.064698)         elapsed: 11.042567 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Analyze the cr yaml] *****************************************************
task path: /utils/fcm/update_litmus_result_resource.yml:38
2020-03-03T07:44:13.672813 (delta: 0.056197)         elapsed: 11.098811 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Apply the litmus result CR] **********************************************
task path: /utils/fcm/update_litmus_result_resource.yml:41
2020-03-03T07:44:13.735644 (delta: 0.062782)         elapsed: 11.161642 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Check if the cspc pool exists] *******************************************
task path: /experiments/chaos/cspc_pod_disruption_budget/test.yml:20
2020-03-03T07:44:13.789083 (delta: 0.053392)         elapsed: 11.215081 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get cspc -n openebs --no-headers -o custom-columns=:metadata.name", "delta": "0:00:00.916791", "end": "2020-03-03 07:44:14.870523", "failed_when_result": false, "rc": 0, "start": "2020-03-03 07:44:13.953732", "stderr": "", "stderr_lines": [], "stdout": "cstor-cspc-disk-pool", "stdout_lines": ["cstor-cspc-disk-pool"]}

TASK [Obtain the number of cspi from CSPC pool config] *************************
task path: /experiments/chaos/cspc_pod_disruption_budget/test.yml:27
2020-03-03T07:44:14.934132 (delta: 1.145003)         elapsed: 12.36013 ******** 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get cspc cstor-cspc-disk-pool -n openebs --no-headers -o yaml | grep 'poolConfig:' |wc -l", "delta": "0:00:00.977818", "end": "2020-03-03 07:44:16.083361", "rc": 0, "start": "2020-03-03 07:44:15.105543", "stderr": "", "stderr_lines": [], "stdout": "3", "stdout_lines": ["3"]}

TASK [Check if the desired number of pool pods are running] ********************
task path: /experiments/chaos/cspc_pod_disruption_budget/test.yml:35
2020-03-03T07:44:16.139418 (delta: 1.205216)         elapsed: 13.565416 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pods -n openebs -l openebs.io/cstor-pool-cluster=cstor-cspc-disk-pool --no-headers | wc -l", "delta": "0:00:00.820299", "end": "2020-03-03 07:44:17.132226", "failed_when_result": false, "rc": 0, "start": "2020-03-03 07:44:16.311927", "stderr": "", "stderr_lines": [], "stdout": "3", "stdout_lines": ["3"]}

TASK [Obtain the list of pool pods] ********************************************
task path: /experiments/chaos/cspc_pod_disruption_budget/test.yml:44
2020-03-03T07:44:17.195240 (delta: 1.055776)         elapsed: 14.621238 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pods -n openebs -l openebs.io/cstor-pool-cluster=cstor-cspc-disk-pool --no-headers -o custom-columns=:.metadata.name", "delta": "0:00:00.880966", "end": "2020-03-03 07:44:18.265953", "rc": 0, "start": "2020-03-03 07:44:17.384987", "stderr": "", "stderr_lines": [], "stdout": "cstor-cspc-disk-pool-f79s-65d8489458-wlwsp\ncstor-cspc-disk-pool-psfz-644897bc85-r8vhf\ncstor-cspc-disk-pool-q8pc-6d89b9d68b-mcwhz", "stdout_lines": ["cstor-cspc-disk-pool-f79s-65d8489458-wlwsp", "cstor-cspc-disk-pool-psfz-644897bc85-r8vhf", "cstor-cspc-disk-pool-q8pc-6d89b9d68b-mcwhz"]}

TASK [Obtain the node list where the pools are scheduled] **********************
task path: /experiments/chaos/cspc_pod_disruption_budget/test.yml:52
2020-03-03T07:44:18.324337 (delta: 1.129052)         elapsed: 15.750335 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pods -n openebs -l openebs.io/cstor-pool-cluster=cstor-cspc-disk-pool --no-headers -o custom-columns=:spec.nodeName", "delta": "0:00:00.891650", "end": "2020-03-03 07:44:19.419874", "rc": 0, "start": "2020-03-03 07:44:18.528224", "stderr": "", "stderr_lines": [], "stdout": "giri-node2\ngiri-node1\ngiri-node3", "stdout_lines": ["giri-node2", "giri-node1", "giri-node3"]}

TASK [include_tasks] ***********************************************************
task path: /experiments/chaos/cspc_pod_disruption_budget/test.yml:60
2020-03-03T07:44:19.482257 (delta: 1.157875)         elapsed: 16.908255 ******* 
included: /chaoslib/kubectl/cordon_drain_node.yaml for 127.0.0.1

TASK [Identify the application node] *******************************************
task path: /chaoslib/kubectl/cordon_drain_node.yaml:3
2020-03-03T07:44:19.594101 (delta: 0.111797)         elapsed: 17.020099 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pod cstor-cspc-disk-pool-f79s-65d8489458-wlwsp -n openebs --no-headers -o custom-columns=:spec.nodeName", "delta": "0:00:00.871758", "end": "2020-03-03 07:44:20.644578", "rc": 0, "start": "2020-03-03 07:44:19.772820", "stderr": "", "stderr_lines": [], "stdout": "giri-node2", "stdout_lines": ["giri-node2"]}

TASK [Record the application node name] ****************************************
task path: /chaoslib/kubectl/cordon_drain_node.yaml:11
2020-03-03T07:44:20.704525 (delta: 1.110338)         elapsed: 18.130523 ******* 
ok: [127.0.0.1] => {"ansible_facts": {"app_node": "giri-node2"}, "changed": false}

TASK [Drain the application node] **********************************************
task path: /chaoslib/kubectl/cordon_drain_node.yaml:15
2020-03-03T07:44:20.796336 (delta: 0.091436)         elapsed: 18.222334 ******* 
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl drain giri-node2 --ignore-daemonsets --delete-local-data --force", "delta": "0:00:32.907790", "end": "2020-03-03 07:44:53.870909", "rc": 0, "start": "2020-03-03 07:44:20.963119", "stderr": "WARNING: the server could not find the requested resource: calico-node-cxk7z, kube-proxy-ckbk7, openebs-cstor-csi-node-l66vf, openebs-ndm-6rvls; Deleting pods with local storage: cstor-cspc-disk-pool-f79s-65d8489458-wlwsp", "stderr_lines": ["WARNING: the server could not find the requested resource: calico-node-cxk7z, kube-proxy-ckbk7, openebs-cstor-csi-node-l66vf, openebs-ndm-6rvls; Deleting pods with local storage: cstor-cspc-disk-pool-f79s-65d8489458-wlwsp"], "stdout": "node/giri-node2 cordoned\npod/calico-node-cxk7z evicted\npod/openebs-ndm-6rvls evicted\npod/openebs-cstor-csi-node-l66vf evicted\npod/kube-proxy-ckbk7 evicted\npod/cstor-cspc-disk-pool-f79s-65d8489458-wlwsp evicted", "stdout_lines": ["node/giri-node2 cordoned", "pod/calico-node-cxk7z evicted", "pod/openebs-ndm-6rvls evicted", "pod/openebs-cstor-csi-node-l66vf evicted", "pod/kube-proxy-ckbk7 evicted", "pod/cstor-cspc-disk-pool-f79s-65d8489458-wlwsp evicted"]}

TASK [Wait for application pod reschedule (evict)] *****************************
task path: /chaoslib/kubectl/cordon_drain_node.yaml:26
2020-03-03T07:44:53.938990 (delta: 33.142582)         elapsed: 51.364988 ****** 
ok: [127.0.0.1] => {"changed": false, "elapsed": 30, "path": null, "port": null, "search_regex": null, "state": "started"}

TASK [Uncordon the application node] *******************************************
task path: /chaoslib/kubectl/cordon_drain_node.yaml:35
2020-03-03T07:45:24.389617 (delta: 30.450579)         elapsed: 81.815615 ****** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Check if the corresponding pool pod is evicted] **************************
task path: /experiments/chaos/cspc_pod_disruption_budget/test.yml:66
2020-03-03T07:45:24.456119 (delta: 0.066449)         elapsed: 81.882117 ******* 
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -n openebs -l openebs.io/cstor-pool-cluster=cstor-cspc-disk-pool --no-headers -o custom-columns=:status.phase | grep Running | wc -l", "delta": "0:00:00.889071", "end": "2020-03-03 07:45:25.541525", "rc": 0, "start": "2020-03-03 07:45:24.652454", "stderr": "", "stderr_lines": [], "stdout": "2", "stdout_lines": ["2"]}

TASK [Obtain the running cstor pool pods] **************************************
task path: /experiments/chaos/cspc_pod_disruption_budget/test.yml:77
2020-03-03T07:45:25.606481 (delta: 1.150277)         elapsed: 83.032479 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pods -n openebs -l openebs.io/cstor-pool-cluster=cstor-cspc-disk-pool -o json | jq -r '.items[] | select(.status.phase == \"Running\" or ([ .status.conditions[] | select(.type == \"Ready\" and .status == \"False\") ] | length ) == 1 ) | .metadata.name'", "delta": "0:00:00.869108", "end": "2020-03-03 07:45:26.657265", "rc": 0, "start": "2020-03-03 07:45:25.788157", "stderr": "", "stderr_lines": [], "stdout": "cstor-cspc-disk-pool-psfz-644897bc85-r8vhf\ncstor-cspc-disk-pool-q8pc-6d89b9d68b-mcwhz", "stdout_lines": ["cstor-cspc-disk-pool-psfz-644897bc85-r8vhf", "cstor-cspc-disk-pool-q8pc-6d89b9d68b-mcwhz"]}

TASK [Obtain the node where one of the above pods is scheduled] ****************
task path: /experiments/chaos/cspc_pod_disruption_budget/test.yml:86
2020-03-03T07:45:26.716267 (delta: 1.109713)         elapsed: 84.142265 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pods -n openebs cstor-cspc-disk-pool-psfz-644897bc85-r8vhf --no-headers -o custom-columns=:spec.nodeName", "delta": "0:00:00.809622", "end": "2020-03-03 07:45:27.693943", "rc": 0, "start": "2020-03-03 07:45:26.884321", "stderr": "", "stderr_lines": [], "stdout": "giri-node1", "stdout_lines": ["giri-node1"]}

TASK [Drain the above node] ****************************************************
task path: /experiments/chaos/cspc_pod_disruption_budget/test.yml:94
2020-03-03T07:45:27.754298 (delta: 1.037984)         elapsed: 85.180296 ******* 
fatal: [127.0.0.1]: FAILED! => {"changed": false, "msg": "async task did not complete within the requested time"}
...ignoring

TASK [Confirm that the pod disruption budget is honored] ***********************
task path: /experiments/chaos/cspc_pod_disruption_budget/test.yml:103
2020-03-03T07:48:01.262263 (delta: 153.50791)         elapsed: 238.688261 ***** 
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -n openebs -l openebs.io/cstor-pool-cluster=cstor-cspc-disk-pool --no-headers -o custom-columns=:status.phase | grep Running|wc -l", "delta": "0:00:01.051064", "end": "2020-03-03 07:48:02.505548", "rc": 0, "start": "2020-03-03 07:48:01.454484", "stderr": "", "stderr_lines": [], "stdout": "2", "stdout_lines": ["2"]}

TASK [set_fact] ****************************************************************
task path: /experiments/chaos/cspc_pod_disruption_budget/test.yml:114
2020-03-03T07:48:02.572029 (delta: 1.309719)         elapsed: 239.998027 ****** 
ok: [127.0.0.1] => {"ansible_facts": {"flag": "Pass"}, "changed": false}

TASK [Obtain the drained node list] ********************************************
task path: /experiments/chaos/cspc_pod_disruption_budget/test.yml:125
2020-03-03T07:48:02.670054 (delta: 0.097929)         elapsed: 240.096052 ****** 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get nodes -o jsonpath=\"{range.items[?(@.spec.unschedulable)]}{.metadata.name}:{end}\" | tr \":\" \"\\n\"", "delta": "0:00:01.067707", "end": "2020-03-03 07:48:03.935060", "rc": 0, "start": "2020-03-03 07:48:02.867353", "stderr": "", "stderr_lines": [], "stdout": "giri-node1\ngiri-node2", "stdout_lines": ["giri-node1", "giri-node2"]}

TASK [Uncordon the nodes] ******************************************************
task path: /experiments/chaos/cspc_pod_disruption_budget/test.yml:133
2020-03-03T07:48:03.994037 (delta: 1.323708)         elapsed: 241.420035 ****** 
changed: [127.0.0.1] => (item=giri-node1) => {"attempts": 1, "changed": true, "cmd": "kubectl uncordon giri-node1", "delta": "0:00:00.855518", "end": "2020-03-03 07:48:05.073082", "item": "giri-node1", "rc": 0, "start": "2020-03-03 07:48:04.217564", "stderr": "", "stderr_lines": [], "stdout": "node/giri-node1 uncordoned", "stdout_lines": ["node/giri-node1 uncordoned"]}
changed: [127.0.0.1] => (item=giri-node2) => {"attempts": 1, "changed": true, "cmd": "kubectl uncordon giri-node2", "delta": "0:00:00.852662", "end": "2020-03-03 07:48:06.106199", "item": "giri-node2", "rc": 0, "start": "2020-03-03 07:48:05.253537", "stderr": "", "stderr_lines": [], "stdout": "node/giri-node2 uncordoned", "stdout_lines": ["node/giri-node2 uncordoned"]}

TASK [include_tasks] ***********************************************************
task path: /experiments/chaos/cspc_pod_disruption_budget/test.yml:144
2020-03-03T07:48:06.173661 (delta: 2.179578)         elapsed: 243.599659 ****** 
included: /utils/fcm/update_litmus_result_resource.yml for 127.0.0.1

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
task path: /utils/fcm/update_litmus_result_resource.yml:3
2020-03-03T07:48:06.263849 (delta: 0.09015)         elapsed: 243.689847 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Analyze the cr yaml] *****************************************************
task path: /utils/fcm/update_litmus_result_resource.yml:14
2020-03-03T07:48:06.317958 (delta: 0.054067)         elapsed: 243.743956 ****** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Apply the litmus result CR] **********************************************
task path: /utils/fcm/update_litmus_result_resource.yml:17
2020-03-03T07:48:06.373095 (delta: 0.055093)         elapsed: 243.799093 ****** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
task path: /utils/fcm/update_litmus_result_resource.yml:27
2020-03-03T07:48:06.447153 (delta: 0.07386)         elapsed: 243.873151 ******* 
changed: [127.0.0.1] => {"changed": true, "checksum": "b0cbe7df784ed3df3b70e257d37330f5502719a0", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "5e1462edd17ca7f784615f3aeb9945db", "mode": "0644", "owner": "root", "size": 451, "src": "/root/.ansible/tmp/ansible-tmp-1583221686.51-232890923166642/source", "state": "file", "uid": 0}

TASK [Analyze the cr yaml] *****************************************************
task path: /utils/fcm/update_litmus_result_resource.yml:38
2020-03-03T07:48:08.236864 (delta: 1.78966)         elapsed: 245.662862 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "cat litmus-result.yaml", "delta": "0:00:00.760668", "end": "2020-03-03 07:48:09.169372", "rc": 0, "start": "2020-03-03 07:48:08.408704", "stderr": "", "stderr_lines": [], "stdout": "---\napiVersion: litmus.io/v1alpha1\nkind: LitmusResult\nmetadata:\n\n  # name of the litmus testcase\n  name: cspc-pod-disruption-budget \nspec:\n\n  # holds information on the testcase\n  testMetadata:\n    app:  \n    chaostype: cspc-pod-disruption-budget \n\n  # holds the state of testcase,  manually updated by json merge patch\n  # result is the useful value today, but anticipate phase use in future \n  testStatus: \n    phase: completed  \n    result: Pass ", "stdout_lines": ["---", "apiVersion: litmus.io/v1alpha1", "kind: LitmusResult", "metadata:", "", "  # name of the litmus testcase", "  name: cspc-pod-disruption-budget ", "spec:", "", "  # holds information on the testcase", "  testMetadata:", "    app:  ", "    chaostype: cspc-pod-disruption-budget ", "", "  # holds the state of testcase,  manually updated by json merge patch", "  # result is the useful value today, but anticipate phase use in future ", "  testStatus: ", "    phase: completed  ", "    result: Pass "]}

TASK [Apply the litmus result CR] **********************************************
task path: /utils/fcm/update_litmus_result_resource.yml:41
2020-03-03T07:48:09.225776 (delta: 0.988867)         elapsed: 246.651774 ****** 
changed: [127.0.0.1] => {"changed": true, "failed_when_result": false, "method": "patch", "result": {"apiVersion": "litmus.io/v1alpha1", "kind": "LitmusResult", "metadata": {"creationTimestamp": "2020-03-03T01:35:23Z", "generation": 4, "name": "cspc-pod-disruption-budget", "resourceVersion": "261548", "selfLink": "/apis/litmus.io/v1alpha1/litmusresults/cspc-pod-disruption-budget", "uid": "33e75fd1-5c14-43cc-9486-2d74fbf02103"}, "spec": {"testMetadata": {"chaostype": "cspc-pod-disruption-budget"}, "testStatus": {"phase": "completed", "result": "Pass"}}}}
META: ran handlers
META: ran handlers

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=28   changed=18   unreachable=0    failed=0   

2020-03-03T07:48:10.079278 (delta: 0.852844)         elapsed: 247.505276 ****** 
=============================================================================== 
```

**Review by shashank855 — APPROVED** [review]
\lgtm

File: experiments/chaos/cspc_pod_disruption_budget/test.yml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+          args:
+            executable: /bin/bash
+          register: pool_count
+          until: pool_count.stdout|int == (cspi_count.stdout|int-1)
+          delay: 10
+          retries: 30
+
+        - name: Obtain the running cstor pool pods
+          shell: >
+            kubectl get pods -n {{ openebs_ns }} -l openebs.io/cstor-pool-cluster={{ cspc_name }} -o json
```
Comment (shashank855) [inline experiments/chaos/cspc_pod_disruption_budget/test.yml:79]: _awesome filtering._

**Review by somesh2905 — APPROVED** [review]
/lgtm

**Pull request merged**